jetty-client/src/main/java/org/eclipse/jetty/client/api/Request.java

```diff
@@ -50,6 +50,22 @@
  */
 public interface Request
 {
+    /**
+     * <p>Returns the connection associated with this request.</p>
+     * <p>The connection is available only starting from the
+     * {@link #onRequestBegin(BeginListener) request begin} event,
+     * when a connection is associated with the request to be sent,
+     * otherwise {@code null} is returned.</p>
+     *
+     * @return the connection associated with this request,
+     * or {@code null} if there is no connection associated
+     * with this request
+     */
+    default Connection getConnection()
```

Comment:
Why mark that method as `default`?

Reply:
Because it is an existing interface that people may have implemented (e.g. for wrapping), and we don't want to break them.